### PR TITLE
fix(cli): local sources are not cached in memory

### LINKED ISSUE #4743: Deno.compile caching/cache-busting not behaving as expected
**test.ts**
```
console.log('Hello')
```

**main.ts**
```
setInterval(async () => {
    const start = +new Date
    const [ diagnostics, emitMap ] = await Deno.compile('./test.ts')
    console.log(`Took ${+new Date - start} ms to compile`)
    console.log(emitMap)
}, 1000)
```

**deno main.ts output**
```
Took 357 ms to compile
{
 file:///Users/jaka/Desktop/deno-compile-caching/test.js.map: "{\"version\":3,\"file\":\"test.js\",\"sourceRoot\":\"\",\"sources\":[\"test.ts\"],\"names\":[],\"mappings\":\";AAAA,OAA...",
 file:///Users/jaka/Desktop/deno-compile-caching/test.js: "\"use strict\";\nconsole.log('Hello');\n//# sourceMappingURL=test.js.map"
}
[... identical output infinitely ...]
```

The `Deno.compile()` always takes 300ms+ and the output does not change even if `test.ts` is changed while the script is running. Curiously, mtime of `test{.ts.js,.ts.js.map,.meta}` in the cache directory are changing.

Expected:
  - When not changed: returns instantly, does not change files cache
  - When changed: takes a while, picks up changes, changes files in cache

Actual:
  - When not changed: takes a while, changes files in cache
  - When changed: takes a while, does not pick up changes, changes files in cache

deno 0.40.0

### LINKED ISSUE #5253: Compiler APIs should allow a `reload` option to cache bust
Currently there is no way to internally cache bust with the compiler APIs, only if Deno is invoked with `--reload` will the compiler APIs cache bust.

### LINKED ISSUE #5631: Deno.bundle uses cache
I'm creating [a script](https://amatiasq.com/deno/web.ts):

```sh
deno --allow-read=. --allow-net https://amatiasq.com/deno/web.ts my-script.ts
```

That watches the file system and bundles `my-script.ts` into a frontend-ready file.

The issue is that when a file system event is detected `Deno.bundle` bundles again from cache. Making the tool unusable.

I wasn't able to find a way to clear cache, I tried starting a process `deno bundle my-script.ts` but that doesn't allow me to pass a custom `tsconfig` to add `lib: ["DOM"]`.

### LINKED ISSUE #6116: bug(Worker): Not loading latest version of specified script
## Steps to reproduce
1. Create a file `script.ts` with the content `console.log('hello')`.
2. Create another file `entry.ts` with the following content:
```ts
while(true) {
  const worker = new Worker('./script.ts', {type: 'module'})
  await new Promise(resolve => setTimeout(resolve, 1000))
  worker.terminate()
}
```
3. Run this command: `deno run --allow-read main.ts`.
4. Modify the content of `script.ts` to `console.log('bye')` and save it.

## Expected result
The subsequent log of `deno run --allow-read main.ts` should be `bye`.

## Actual result
After step 4, the subsequent log of  `deno run --allow-read main.ts` is still `hello`.

## Environment:
```
macOS Catalina (v10.15.4)
```
```
$ deno --version
deno 1.0.5
v8 8.4.300
typescript 3.9.2
```

## Is this introduced lately?
No, this is happening on every version of Deno since `v1.0.0` until `v1.0.5`.

## Current workaround
Refer https://stackoverflow.com/a/41790024/6587634

### PR DESCRIPTION
**Pull request by kitsonk**
Fixes #4743
Closes #5253
Fixes #5631
Fixes #6116

### CODE REVIEW

File: cli/file_fetcher.rs [hunk 1]
```diff
@@ -457,36 +457,55 @@ impl FileFetcher {
...
     }
   }
 
   /// Get a previously in memory cached file.
-  pub fn get_cached(&self, specifier: &ModuleSpecifier) -> Option<File> {
-    self.cache.get(specifier)
-  }
+  // pub fn get_cached(&self, specifier: &ModuleSpecifier) -> Option<File> {
+  //   self.cache.get(specifier)
+  // }
```
Comment (ry) [inline cli/file_fetcher.rs:484]: Remove

**Review by ry — COMMENTED** [review]
LGTM - nice work. I'll wait for @bartlomieju to give approval.

**Review by bartlomieju — COMMENTED** [review]
Looks good, but I'd like to a see an integration test that uses one of the APIs that had cache busting problem before landing this PR.

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/tests/integration_tests.rs [hunk 2]
```diff
@@ -426,6 +426,54 @@ fn cache_invalidation_test_no_check() {
...
+      { type: "module" },
+    );
+    await new Promise((res) => setTimeout(res, 3000));
+    worker_a.terminate();
+    const worker_b = new Worker(
+      new URL("fixture_b.ts", import.meta.url).href,
+      { type: "module" },
+    );
+    await new Promise((res) => setTimeout(res, 3000));
+    worker_b.terminate();
```
Comment (bartlomieju) [inline cli/tests/integration_tests.rs:445]: Maybe write fixtures to disk in this script?

Comment (kitsonk) [reply cli/tests/integration_tests.rs:445]: 🤦  yes that makes a huge amount of sense!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -31,8 +31,9 @@ jobs:
       # e.g. a flaky test.
       # Don't fast-fail on tag build because publishing binaries shouldn't be
       # prevented if 'cargo publish' fails (which can be a false negative).
-      fail-fast: ${{ github.event_name == 'pull_request' || (github.ref !=
-        'refs/heads/master' && !startsWith(github.ref, 'refs/tags/')) }}
+      # fail-fast: ${{ github.event_name == 'pull_request' || (github.ref !=
+      #   'refs/heads/master' && !startsWith(github.ref, 'refs/tags/')) }}
+      fail-fast: false
```
Comment (bartlomieju) [inline .github/workflows/ci.yml:36]: Temporarily added to debug CI problem, revert before merging

**Review by bartlomieju — CHANGES_REQUESTED** [review]
Note to self:

**Comment by kitsonk** [comment]
Good to know that it just isn't me. 🤣

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**